src/components/ListLayout.jsx

```diff
@@ -61,6 +61,71 @@ const ListLayout = ({ items, localToken }) => {
     return timeCheck;
   }
 
+  // updates isActive property of item to true if item has 2+ purchases and has been purchased within calculated estimate
+  // isActive is defaulted to false when item is added
+  useEffect(() => {
+    items.forEach((item) => {
+      const dateOfLastTransaction =
+        item.totalPurchases > 0 ? item.purchasedDate : item.createdAt;
+      const daysSinceLastTransaction =
+        (currentTime - dateOfLastTransaction) / oneDay;
+      if (
+        item.totalPurchases > 1 &&
+        daysSinceLastTransaction < 2 * item.previousEstimate
+      ) {
+        let itemToUpdate = {
+          isActive: true,
+        };
+        setUpdateToDb(localToken, item.id, itemToUpdate);
+      }
+    });
+    //suggested dependency array via React and I agree with the suggestion if anyone has thoughts on this please let me know!
+  }, [items, localToken, currentTime]);
+
+  //filters items to only display items a user is searching by via the input bar
+  const filteredItems = items.filter((item) =>
+    item.id.includes(filter.toLowerCase()),
+  );
+
+  const groups = [
+    {
+      label: 'Soon',
+      sublabel: "We think you'll need this in less than 7 days",
+      groupFilter: (item) => {
+        return item.previousEstimate < 7 && item.isActive === true;
+      },
+      colorClass: 'bg-rose-100',
+    },
+    {
+      label: 'Kind of soon',
+      sublabel: "We think you'll need this in less than 30 days",
+      groupFilter: (item) => {
```

Comment:
I notice that few days after it's more obvious which names are more ambiguous: groupFilter would be more accurate with something like isItemInGroup (result is boolean, and function tells you if items belongs to that group... word filter feels unnecessary but I'm not 100% sure about that one)